geocode-offline/src/main/java/com/esri/arcgismaps/sample/geocodeoffline/MainActivity.kt

```diff
@@ -0,0 +1,267 @@
+
+    // display the metro area of the tapped location
+    private val descriptionTV by lazy {
+        activityMainBinding.descriptionTV
+    }
+
+    private val provisionPath: String by lazy {
+        getExternalFilesDir(null)?.path.toString() + File.separator + getString(R.string.app_name)
+    }
+
+    // create a picture marker symbol
+    private val pinSymbol: PictureMarkerSymbol by lazy {
+        createPinSymbol()
+    }
+
+    // geocode parameters used to perform a search
+    private val geocodeParameters: GeocodeParameters by lazy {
+        GeocodeParameters().apply {
+            // get all attributes
+            resultAttributeNames.add("*")
+            // get only the closest result
+            maxResults = 1
+        }
+    }
+
+    // locator task to provide geocoding services
+    private val locatorTask: LocatorTask by lazy {
+        LocatorTask(File(provisionPath, getString(R.string.san_diego_loc)).path)
+    }
+
+    // create a graphics overlay
+    private val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // load the tile cache from local storage
+        val tileCache = TileCache("$provisionPath/streetmap_SD.tpkx")
+        // create a tiled layer and add it to as the base map
+        val tiledLayer = ArcGISTiledLayer(tileCache)
+        mapView.apply {
+            map = ArcGISMap(Basemap(tiledLayer))
+            // set map initial viewpoint
+            map?.initialViewpoint = Viewpoint(32.72, -117.155, 120000.0)
+            // add a graphics overlay to the map view
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // load geocode locator task
+        lifecycleScope.launch {
+            locatorTask.load().onSuccess {
+                mapView.onSingleTapConfirmed.collect { event ->
+                    // find address with reverse geocode using the tapped location
+                    event.mapPoint?.let { mapPoint -> findAddressReverseGeocode(mapPoint) }
+                }
+            }.onFailure {
+                showError(it.message.toString())
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu, menu)
+        val search = menu.findItem(R.id.appSearchBar)
+        // set up address search view and listeners
+        setupAddressSearchView(search.actionView as SearchView)
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    /**
+     * Sets up the address SearchView and uses MatrixCursor to
+     * show suggestions to the user as text is entered.
+     */
+    private fun setupAddressSearchView(addressSearchView: SearchView) {
```

Comment:
```suggestion
    private fun setupAddressSearchView(addressSearchView: SearchView) {
        // disable threshold to show results from single character
        val autoCompleteTextViewID = resources.getIdentifier("search_src_text", "id", packageName)
        addressSearchView.findViewById<AutoCompleteTextView>(autoCompleteTextViewID).threshold = 0
```

@hud10837 It seems like the two characters needed for the search result is an Android `SearchView` behavior. The default value for the threshold is for `android.widget.AutoCompleteTextView.getThreshold()` is `1`. (value of the limit of characters needed to display results). 

Although I cannot confirm that from the [Android docs](https://developer.android.com/reference/android/widget/AutoCompleteTextView#getThreshold()), upon manually setting the `threshold` value to 0, the search results show up on a single character! 

That being said, I don't think I would be in favor of adding this change, since it's not a Map SDK/sample issue and I think the default threshold used for producing search results is set for a reason. Probably to save on resources used.